dotCMS/src/main/java/com/dotcms/rest/api/v1/folder/FolderHelper.java

```diff
@@ -25,93 +24,58 @@
-            return folders
-                    .stream()
-                    .collect(Collectors.toList());
-        }
-
-        protected void setFolders(List<Folder> folders) {
-            this.folders = folders;
-        }
-
-        protected List<ErrorEntity> getErrorEntities() {
-            List<ErrorEntity> ret = new ArrayList<ErrorEntity>();
-            for (Map<String,String> error : errors) {
-                ErrorEntity ee = new ErrorEntity(error.get("failedPath"),error.get("errorMessage"));
-                ret.add(ee);
-            }
-            return ret;
-        }
-
-        protected void setErrors(List<Map<String, String>> errors) {
-            this.errors = errors;
-        }
-    }
-
     /**
      * Get the instance.
      *
-     * @return BrowserTreeHelper
+     * @return FolderHelper
      */
     public static FolderHelper getInstance() {
 
         return FolderHelper.SingletonHolder.INSTANCE;
     } // getInstance.
 
-    public FolderResults createFolders(List<String> paths, String siteName, User user) throws DotDataException, DotSecurityException {
+    /**
+     * Creates all folders and subfolders passed in the list.
+     *
+     * @param paths List of folders to create
+     * @param siteName siteName where the folders are gonna be created
+     * @param user user to create folders
+     * @return list of folders created
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    public List createFolders(final List<String> paths, final String siteName, final User user) throws DotDataException, DotSecurityException {
 
-        List<Folder> savedFolders = new ArrayList<Folder>();
-        List<Map<String, String>> errors = new ArrayList<Map<String, String>>();
-        Host host = hostAPI.findByName(siteName, user, true);
+        final List<Folder> savedFolders = new ArrayList<Folder>();
+        final Host host = hostAPI.findByName(siteName, user, true);
+        if(!UtilMethods.isSet(host)) {
+            throw new IllegalArgumentException(String.format(" Couldn't find any host with name `%s` ",siteName));
+        }
 
-        for (String path : paths) {
-            try {
-                savedFolders.add(folderAPI.createFolders(path, host, user, true));
-            } catch (Exception e) {
-                Logger.error(this, e.getMessage(), e);
-                Map<String, String> error = new HashMap<String, String>();
-                error.put("failedPath",path);
-                error.put("errorMessage",e.getMessage());
-                errors.add(error);
-            }
+        for (final String path : paths) {
+            savedFolders.add(folderAPI.createFolders(path, host, user, true));
         }
-        return new FolderResults(savedFolders, errors);
+        return savedFolders.stream().map(folder -> {
+                        try {
+                            return folder.getMap();
+                        } catch (Exception e) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5552986)